tests/integration/api.int.js

```diff
@@ -90,3 +114,143 @@ describe('API proxy endpoint integration tests', () => {
 		});
 	});
 });
+
+/*
+ * BEGIN 'POST' TESTS
+ */
+const mockPostPayload = {
+	queries,
+};
+
+const runMutationTest = ({ csrfHeaders, method, test, queries }) => server => {
+	csrfHeaders = csrfHeaders || getCsrfHeaders;
+	return csrfHeaders()
+		.then(([headerToken, cookieToken]) => {
+			const headers = {
+				'x-csrf-jwt': headerToken,
+				'content-type': 'application/x-www-form-urlencoded',
+				cookie: `x-csrf-jwt=${cookieToken}`,
+			};
+			const request = {
+				method,
+				url: `/mu_api${queries ? `?queries=${queries}` : ''}`,
+				payload: queries
+					? undefined
+					: querystring.stringify(mockPostPayload),
+				credentials: 'whatever',
+				headers,
+			};
+			return server.inject(request);
+		})
+		.then(test)
+		.then(() => server.stop())
+		.catch(err => {
+			server.stop();
+			throw err;
+		});
+};
+
+describe('API proxy POST endpoint integration tests', () => {
```

Comment:
General comment these look all like positive outcome tests, should we have negative tests. This is a bit different than unit tests but just commenting on that observation